### DISCUSSION TITLE
Aggregation with user defined function

### DISCUSSION THREAD

**Discussion by JeremyBloom** [dflib/dflib#339] [Q&A]
I have a dataframe grouped by a set of key columns. I want to aggregate the the values in another, data column using a user defined function (the type of the data column is not a primitive or boxed primitive like a number; the aggregation is specific to the type of the data). How do I specify the argument of the agg method to call my user defined function? I don't see this case addressed in the documentation and I can't make it out in the code. Thanks.

**Comment by andrus** [comment]
Hi Jeremy, good question! We have a flavor of custom aggregation where you'd start an aggregation expression by selecting a column, and then pass a custom function that takes a Series and produces a value:

```java
df.group("a", "b").agg(
    $col("c").agg(series -> series.size());
```

Will that work on your case?
